src/java.base/share/classes/java/lang/invoke/ClassSpecializer.java

```diff
@@ -993,16 +931,13 @@ protected Factory makeFactory() {
     // Other misc helpers:
     private static final String MH = "java/lang/invoke/MethodHandle";
     private static final String MH_SIG = "L" + MH + ";";
-    private static final String STABLE = "jdk/internal/vm/annotation/Stable";
-    private static final String STABLE_SIG = "L" + STABLE + ";";
-    private static final String[] E_THROWABLE = new String[] { "java/lang/Throwable" };
     static {
         assert(MH_SIG.equals(classSig(MethodHandle.class)));
         assert(MH.equals(classBCName(MethodHandle.class)));
     }
 
-    static String methodSig(MethodType mt) {
-        return mt.toMethodDescriptorString();
+    static MethodTypeDesc methodSig(MethodType mt) {
+        return MethodTypeDesc.ofDescriptor(mt.toMethodDescriptorString());
```

Comment:
`MethodType.describeConstable` is a bit slow for its stream-based implementation, but we should still favor conversion of individual parameter type and return type than using descriptors, which is slower to parse per my benchmark in #13186.